src/link/MachO/Dylib.zig

```diff
@@ -179,12 +182,39 @@ pub fn parseSymbols(self: *Dylib) !void {
             },
             .dylib = self,
         };
+        log.warn("Here {s}", .{name});
 
         try self.symbols.putNoClobber(self.allocator, name, &proxy.base);
     }
 }
 
 pub fn isDylib(file: fs.File) !bool {
+    {
+        const fat_header = try file.reader().readStructEndian(macho.fat_header, builtin.Endian.Big);
+        if(fat_header.magic == macho.FAT_MAGIC) {
+            log.warn("Num Arches = {}", .{fat_header.nfat_arch});
+            var fat_arch_index: u32 = 0;
+            while (fat_arch_index < fat_header.nfat_arch) : (fat_arch_index += 1) {
+                const fat_arch = try file.reader().readStructEndian(macho.fat_arch, builtin.Endian.Big);
+                log.warn("Fat arch header details {}, {}, {}, {}", .{
+                    fat_arch.cputype,
+                    fat_arch.cpusubtype,
+                    fat_arch.offset,
+                    fat_arch.size
+                });
+                if(fat_arch.cputype == macho.CPU_TYPE_X86_64) {
```

Comment:
As implied by @LemonBoy, we should leave the decision making whether this FAT dylib contains the architecture of interest to the actual parsing logic, and then warn the user if it doesn't but don't trigger a hard error. I guess a hard error should only ever occur if the parsing cannot be performed, read, the dylib is malformed.

Reply:
At the moment if you try to link with a dylib that doens't match the requested architecture you _do_ get a hard error - so I'm going to match that existing logic for now, but can change both to a warning if you think that would be best afterwards.

Reply:
So I've done this work now - the only potential concern is that we are now reading the fat headers multiple times from disk (once to check if it embeds dylibs & once to search for a good candidate when parsing), I think it would be preferable to do this work only once (especially for people not compiling on an SSD).

Reply:
Are you happy if I leave things as is currently with the PR, or should we address these concerns?